app/models/host.rb

```diff
@@ -86,6 +90,14 @@ class Host < ApplicationRecord
     left_outer_joins(:test_result_profiles)
   }
 
+  scope :with_failed_rules_count, lambda {
+    sq = Host.left_outer_joins(test_result_profiles: :rule_results)
+             .where(rule_results: { result: RuleResult::FAILED })
+             .select(Host.arel_table[:id].as('id'), RuleResult.arel_table[:result].count.as('rules_failed'))
+             .group('hosts.id')
+    joins("INNER JOIN (#{sq.to_sql}) sq ON sq.id = hosts.id")
```

Comment:
So you don't need to select anything? NICE!

Reply:
Right, it works without it 👍